p2panda-net/src/engine/engine.rs

```diff
@@ -641,23 +649,43 @@ impl TopicMap {
         inner.joined.contains(topic) || inner.pending_joins.contains(topic)
     }
 
-    /// Handle incoming messages from gossip overlay or sync session.
+    /// Handle incoming messages from gossip.
     ///
     /// This method forwards messages to the subscribers for the given topic.
-    pub async fn on_message(
+    pub async fn on_gossip_message(
         &self,
         topic: TopicId,
         bytes: Vec<u8>,
         delivered_from: PublicKey,
     ) -> Result<()> {
         let inner = self.inner.read().await;
-        let from_network_tx = inner.earmarked.get(&topic).context("on_message")?;
-        from_network_tx.send(FromNetwork::Message {
+        let from_network_tx = inner.earmarked.get(&topic).context("on_gossip_message")?;
+        from_network_tx.send(FromNetwork::GossipMessage {
             bytes,
             delivered_from: to_public_key(delivered_from),
         })?;
         Ok(())
     }
+
+    /// Handle incoming messages from sync.
+    ///
+    /// This method forwards messages to the subscribers for the given topic.
+    pub async fn on_sync_message(
+        &self,
+        topic: TopicId,
+        header: Vec<u8>,
+        payload: Option<Vec<u8>>,
+        delivered_from: PublicKey,
+    ) -> Result<()> {
+        let inner = self.inner.read().await;
+        let out_tx = inner.earmarked.get(&topic).context("on_sync_message")?;
```

Comment:
Can we rename `out_tx` to `from_network_tx` to keep parity with naming conventions in `on_gossip_message()`?